src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/BlobManager.java

```diff
@@ -222,4 +242,16 @@ private void logAcquireLeaseError(String zipFilename, String containerName, Exce
             exception
         );
     }
+
+    private boolean readyToAcquireLease(CloudBlockBlob cloudBlockBlob) {
```

Comment:
BlobManager is already a little too big. I suggest to introduce LeaseManager component for this method and then move some other lines related to acquiring and releasing lease there. It will also make BlobManager better testable.

Reply:
Refactoring BlobManager involves more changes than the actual fix and may introduce new issues. I think we should leave refactoring to new PR / Jira ticket

Reply:
I agree with this because of urgency of this task.